src/components/ArticleList.jsx

```diff
@@ -8,26 +8,70 @@ function ArticleList() {
   const [articles, setArticles] = useState([]);
   const [isLoading, setIsLoading] = useState(true);
   const { topic } = useParams();
+  const [sort_by, setSort_by] = useState("created_at");
+  const [order, setOrder] = useState("DESC");
 
   useEffect(() => {
     setIsError(false);
-    setIsLoading(true);
-    getAllArticles(topic)
-      .then((response) => {
-        setArticles(response.data.articles);
-        setIsLoading(false);
-      })
-      .catch((error) => {
-        setIsError(true);
-        setIsLoading(false);
+    if (sort_by === "comment_count") {
+      setArticles(sortByComments(order, articles));
+    } else {
+      setIsLoading(true);
+      getAllArticles(topic, sort_by, order)
+        .then((response) => {
+          setArticles(response.data.articles);
+          setIsLoading(false);
+        })
+        .catch((error) => {
+          setIsError(true);
+          setIsLoading(false);
+        });
```

Comment:
This is great!
It comes down to personal preference but I wonder whether it might be better to have a more subtle loading state on this one that doesn't remove all the content from the screen. The momentary loss of the article list isn't as smooth on the eye, it might be possible to have some loading notification render just to the side of the drop down ? Up to you though you might prefer it as it is